onnx/backend/test/case/node/rmsnormalization.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import numpy as np
+
+import onnx
+from onnx.backend.test.case.base import Base
+from onnx.backend.test.case.node import expect
+
+
+# RMS normalization's reference implementation
+def _rms_normalization(X, W, axis=-1, epsilon=1e-5):  # type: ignore
+    X_shape = X.shape
+    X_rank = len(X_shape)
+    if axis < 0:
+        # If axis = -1 and rank of X is 4,
+        # the axis is changed to -1 + 4 = 3,
+        # which means the last axis.
+        axis = axis + X_rank
+    unsqueezed_rank = X_rank - axis
+    reduction_shape = X_shape[0:axis] + (1,) * unsqueezed_rank
+
+    # Parameter used to convert N-D tensor layer
+    # normalization to equivalent 2-D matirx operations.
+    row_number = 1
+    col_number = 1
+    for i in range(X_rank):
+        if i < axis:
+            row_number *= X_shape[i]
+        else:
+            col_number *= X_shape[i]
```

Comment:
Is there a way to vectorize and avoid the loop?

Reply:
I think `row_number = np.prod(X_Shape[:axis]` and `col_number = np.prod(X_shape[axis:])` might work?